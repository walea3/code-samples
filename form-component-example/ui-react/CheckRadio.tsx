import cx from 'classnames';
import React, { ReactNode } from 'react';

import { getDataRef } from '@nel-ui/js-helpers';
import { CheckRadioIcon } from './CheckRadioIcon';
import FormStyles from './FormStyles';

// TODO: Remove inline here when support is added
export type CheckRadioSingular = Omit<CheckRadioProps, 'inline' | 'variant'>;

export interface CheckRadioProps extends React.InputHTMLAttributes<HTMLInputElement> {
  id: string;
  label: ReactNode;
  variant: 'checkbox' | 'radio';
  alert?: 'warning' | 'error';
  inline?: boolean;
  disabled?: boolean;
  'data-ref'?: string;
}

export function CheckRadio({
  'data-ref': dataRef,
  variant,
  label,
  alert,
  inline,
  className,
  ...props
}: CheckRadioProps) {
  return (
    <FormStyles>
      {({ classes }) => (
        <div
          className={cx(classes.checkRadioContainer, {
            [classes.inlineCheckRadio]: inline,
          })}
          data-ref={getDataRef(dataRef, `${variant}Container`)}
        >
          <input
            data-ref={getDataRef(dataRef, `${variant}Input`)}
            type={variant}
            className={classes.checkRadioInput}
            aria-invalid={!!alert}
            {...props}
          />

          <label
            data-ref={getDataRef(dataRef, variant)}
            htmlFor={props.id}
            className={cx(className, classes.checkRadio, classes[variant], {
              [classes[`${alert}CheckRadio`]]: alert,
            })}
          >
            <span
              data-ref={getDataRef(dataRef, `${variant}Control`)}
              className={classes.checkRadioControl}
            >
              <CheckRadioIcon variant={variant} />
            </span>
            {label}
          </label>
        </div>
      )}
    </FormStyles>
  );
}
