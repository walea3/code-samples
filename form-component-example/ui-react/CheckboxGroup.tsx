import cx from 'classnames';
import React, { ChangeEvent, ReactText, useEffect, useState } from 'react';

import { Checkbox, CheckRadioSingular, getIntersectionByValue } from './';
import FormStyles from './FormStyles';

type DivAttr = React.HTMLAttributes<HTMLDivElement>;
type InputEvent = ChangeEvent<HTMLInputElement>;
export type CheckboxGroupResponseItem = number[];

type CheckboxGroupItem = CheckRadioSingular & {
  value: ReactText;
};

export interface CheckboxGroupProps extends Omit<DivAttr, 'onChange'> {
  name: string;
  inline?: boolean;
  className?: string;
  'data-ref'?: string;
  value?: Array<CheckboxGroupItem['value']>;
  items: CheckboxGroupItem[];
  onChange(
    clickEvent: InputEvent,
    checkboxesSelected: {
      selectedItems: CheckRadioSingular[];
      selectedIndexes: CheckboxGroupResponseItem;
    }
  ): void;
}

export function CheckboxGroup({
  name,
  className,
  inline,
  'data-ref': dataRef,
  items,
  value,
  onChange,
  ...props
}: CheckboxGroupProps) {
  if (!items || (items && !items.length)) return null;

  const initialValues = value ? getIntersectionByValue(items, value) : [];

  const nextOptions = items.map(({ id, ...item }, index) => ({
    id,
    checked: initialValues[index] || !!item.checked,
    ...item,
  }));

  const [options, setOptions] = useState(nextOptions);

  useEffect(() => setOptions(nextOptions), [items]);

  const handleChange = (selectedIndex: number) => (event: InputEvent): void => {
    event.stopPropagation();

    const toggleOptions = options.map((checkbox, i) => ({
      ...checkbox,
      checked: i === selectedIndex ? event.target.checked : checkbox.checked,
    }));

    setOptions(toggleOptions);

    const checkboxesSelected = getOnChangeValues(toggleOptions);

    onChange(event, checkboxesSelected);
  };

  const getOnChangeValues = (array: CheckboxGroupProps['items']) => {
    const getIndexes = array.reduce(
      (acc: number[], { checked }, optionIndex) =>
        checked ? [...acc, optionIndex] : acc,
      []
    );

    return {
      selectedItems: array.filter(({ checked }) => checked),
      selectedIndexes: getIndexes,
    };
  };

  const displayMobileInline = items.length <= 2;

  return (
    <FormStyles>
      {({ classes }) => (
        <div className={cx(classes.checkRadioGroup, className)} {...props}>
          <div
            className={cx(classes.checkRadioGroupInner, {
              [classes.largeInlineGroup]: inline && !displayMobileInline,
              [classes.mobileInlineGroup]: inline && displayMobileInline,
            })}
          >
            {options.map((checkbox, i) => (
              <Checkbox
                key={checkbox.id}
                data-ref={dataRef ? `${dataRef}.${i}` : undefined}
                name={name}
                {...checkbox}
                onChange={handleChange(i)}
                inline={inline}
              />
            ))}
          </div>
        </div>
      )}
    </FormStyles>
  );
}
