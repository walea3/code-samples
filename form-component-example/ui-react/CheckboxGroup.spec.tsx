import { mountWithNEL, renderWithNEL } from '@nel-ui/tooling-react/test-utils';
import React, { useEffect, useState } from 'react';
import { CheckboxGroup, CheckboxGroupProps } from './CheckboxGroup';
import { clickCheckbox, findCheckbox } from './CheckRadio.test-utils';

describe('<CheckboxGroup />', () => {
  const checkboxStubData = (valueIsString: boolean) =>
    Array.from({ length: 3 }, (_, i) => ({
      id: `checkbox-test-${i}`,
      value: valueIsString ? `checkbox-test-${i}` : i,
      label: `Item ${i}`,
    }));

  const requiredProps = {
    name: 'test-name',
    items: checkboxStubData(true),
    onChange: jest.fn(),
  };

  const simulatedEventProps = {
    stopPropagation: jest.fn(),
  };

  beforeEach(jest.clearAllMocks);

  const renderComponent = (props: Partial<CheckboxGroupProps> = {}) =>
    renderWithNEL(<CheckboxGroup {...requiredProps} {...props} />);

  const mountComponent = (props: Partial<CheckboxGroupProps> = {}) =>
    mountWithNEL(<CheckboxGroup {...requiredProps} {...props} />);

  describe('@renders', () => {
    test.each`
      name                              | props
      ${'as default'}                   | ${{}}
      ${'with items value as number'}   | ${{ items: checkboxStubData(false) }}
      ${'with no items'}                | ${{ items: [] }}
      ${'with className'}               | ${{ className: 'custom-className' }}
      ${'with inline'}                  | ${{ inline: true }}
      ${'with value as single entry'}   | ${{ value: ['checkbox-test-0'] }}
      ${'with value as multiple entry'} | ${{ value: ['checkbox-test-0', 'checkbox-test-1'] }}
      ${'with data-ref'}                | ${{ 'data-ref': 'uid' }}
      ${'with other props'}             | ${{ 'data-custom': 'uid' }}
    `('$name', ({ props }) => {
      expect(renderComponent(props)).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    beforeEach(() => {
      expect(requiredProps.onChange).toHaveBeenCalledTimes(0);
    });

    describe('should stopPropagation on event', () => {
      beforeEach(() => {
        expect(simulatedEventProps.stopPropagation).toHaveBeenCalledTimes(0);
      });

      test.each`
        name                | selectOrder
        ${'single item'}    | ${[0]}
        ${'multiple items'} | ${[1, 2]}
      `('$name', ({ selectOrder }) => {
        const { root, wrapper } = mountComponent();

        selectOrder.forEach((checkboxIndex) =>
          clickCheckbox(wrapper, checkboxIndex, simulatedEventProps)
        );

        root.unmount();

        expect(simulatedEventProps.stopPropagation).toHaveBeenCalledTimes(
          selectOrder.length
        );
      });
    });

    describe('should call onChange()', () => {
      test.each`
        name                              | selectOrder
        ${'should select one item'}       | ${[0]}
        ${'should select multiple items'} | ${[1, 2]}
      `('$name', ({ selectOrder }) => {
        const { root, wrapper } = mountComponent();

        selectOrder.forEach((checkboxIndex) => clickCheckbox(wrapper, checkboxIndex));

        root.unmount();

        // Note - returns array with item + expected checked state
        const expectedArray = selectOrder.map((orderIndex) => ({
          ...requiredProps.items[orderIndex],
          checked: true,
        }));

        expect(requiredProps.onChange).toHaveBeenCalledTimes(selectOrder.length);
        expect(requiredProps.onChange).toHaveBeenCalledWith(
          expect.objectContaining({ type: 'change' }),
          {
            selectedItems: expectedArray,
            selectedIndexes: selectOrder,
          }
        );
      });
    });

    test('allow items checked status to be managed externally and internally', () => {
      const getCheckedItems = (labels, checkedValues) =>
        checkedValues.map((checked, i) => {
          const id = `${i}`;
          return { id, ...(labels ? { label: id } : {}), checked };
        });

      const CheckboxGroupWrapper = () => {
        const [items, setItems] = useState(getCheckedItems(true, [true, true]));
        useEffect(() => {
          // externally uncheck the second item
          setItems(getCheckedItems(true, [true, false]));
        }, []);
        return <CheckboxGroup {...requiredProps} items={items} />;
      };

      const { root } = mountWithNEL(<CheckboxGroupWrapper />);
      const checkbox = findCheckbox(root, 0);
      root.render();
      // uncheck the first item
      checkbox.simulate('change', { target: { checked: false } });

      expect(requiredProps.onChange).toHaveBeenCalledTimes(1);
      // expect both items to be unchecked
      expect(requiredProps.onChange).toHaveBeenCalledWith(
        expect.objectContaining({ type: 'change' }),
        {
          selectedItems: [],
          selectedIndexes: [],
        }
      );
    });
  });
});
