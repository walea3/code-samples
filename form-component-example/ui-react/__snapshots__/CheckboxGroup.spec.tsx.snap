// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxGroup /> @renders as default 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with className 1`] = `
<div
  class="nel-checkRadioGroup custom-className"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with data-ref 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="uid.0.checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="uid.0.checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="uid.0.checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="uid.0.checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="uid.1.checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="uid.1.checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="uid.1.checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="uid.1.checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="uid.2.checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="uid.2.checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="uid.2.checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="uid.2.checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with inline 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner nel-largeInlineGroup"
  >
    <div
      class="nel-checkRadioContainer nel-inlineCheckRadio"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer nel-inlineCheckRadio"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer nel-inlineCheckRadio"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with items value as number 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with no items 1`] = `null`;

exports[`<CheckboxGroup /> @renders with other props 1`] = `
<div
  class="nel-checkRadioGroup"
  data-custom="uid"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with value as multiple entry 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        checked=""
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        checked=""
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> @renders with value as single entry 1`] = `
<div
  class="nel-checkRadioGroup"
>
  <div
    class="nel-checkRadioGroupInner"
  >
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        checked=""
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-0"
        name="test-name"
        type="checkbox"
        value="checkbox-test-0"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-0"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 0
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-1"
        name="test-name"
        type="checkbox"
        value="checkbox-test-1"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-1"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 1
      </label>
    </div>
    <div
      class="nel-checkRadioContainer"
      data-ref="checkboxContainer"
    >
      <input
        aria-invalid="false"
        class="nel-checkRadioInput"
        data-ref="checkboxInput"
        id="checkbox-test-2"
        name="test-name"
        type="checkbox"
        value="checkbox-test-2"
      />
      <label
        class="nel-checkRadio nel-checkbox"
        data-ref="checkbox"
        for="checkbox-test-2"
      >
        <span
          class="nel-checkRadioControl"
          data-ref="checkboxControl"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 28 28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5,7.4l-8,7.9l-1.9-1.9l-2.1,2.1l4,4.1L21.6,9.5L19.5,7.4z"
            />
          </svg>
        </span>
        Item 2
      </label>
    </div>
  </div>
</div>
`;
